server/src/internalClusterTest/java/org/opensearch/remotemigration/DocrepToRemoteDualReplicationIT.java

```diff
@@ -0,0 +1,420 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.remotemigration;
+
+import org.opensearch.action.admin.indices.stats.CommonStats;
+import org.opensearch.action.admin.indices.stats.ShardStats;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.node.DiscoveryNodes;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.allocation.command.MoveAllocationCommand;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.index.remote.RemoteSegmentStats;
+import org.opensearch.index.seqno.RetentionLeases;
+import org.opensearch.test.InternalTestCluster;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0, autoManageMasterNodes = false)
+public class DocrepToRemoteDualReplicationIT extends MigrationBaseTestCase {
+    private String REMOTE_PRI_DOCREP_REP = "remote-primary-docrep-replica";
+    private String REMOTE_PRI_DOCREP_REMOTE_REP = "remote-primary-docrep-remote-replica";
+    private String FAILOVER_REMOTE_TO_DOCREP = "failover-remote-to-docrep";
+
+    /*
+    Scenario:
+    - Starts 2 docrep backed node
+    - Creates index with 1 replica
+    - Index some docs
+    - Start 1 remote backed node
+    - Move primary copy from docrep to remote through _cluster/reroute
+    - Index some more docs
+    - Assert primary-replica consistency
+     */
+    public void testRemotePrimaryDocRepReplica() throws Exception {
+        internalCluster().setBootstrapClusterManagerNodeIndex(0);
+        internalCluster().startClusterManagerOnlyNode();
+
+        logger.info("---> Starting 2 docrep data nodes");
+        internalCluster().startDataOnlyNode();
+        internalCluster().startDataOnlyNode();
```

Comment:
nit - pls use `startDataOnlyNodes` method for starting 2 or more data only nodes.